sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -113,7 +113,6 @@ public ReactorConnection(String connectionId, ConnectionOptions connectionOption
                         operationTimeout), handler.getErrorContext())));
                 return activeEndpoint.thenReturn(reactorConnection);
             })
-            .or(onClosedError("Could not get active connection."))
```

Comment:
This gave the downstream a non-transient error so retry wouldn't happen.

Reply:
Ahh, good catch. Would it be better to update the onClosedError to return an AmqpException that is transient? I had this error here because if the connection was closed while we were trying to get a new consumer or producer, it would immediately error downstream instead of waiting for a timeout.

Reply:
I changed below `this.endpointStates = this.handler.getEndpointStates()...` to raise an error, which is transient. I think that error will also immediately error down. Do I understand correctly?